storage: TestRangeTransferLeaseExpirationBased failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/01ee0704865391599abef3bbc89f462117f8007a

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestRangeTransferLeaseExpirationBased PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1445911&tab=buildLog

```
=== RUN   TestRangeTransferLeaseExpirationBased
--- FAIL: TestRangeTransferLeaseExpirationBased (0.57s)
=== RUN   TestRangeTransferLeaseExpirationBased/DrainTransfer
I190820 06:51:19.943095 430128 gossip/gossip.go:394  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:36493" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W190820 06:51:19.980699 430128 gossip/gossip.go:1517  [n2] no incoming or outgoing connections
I190820 06:51:19.980785 430128 gossip/gossip.go:394  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:36593" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I190820 06:51:19.983001 430452 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:36493
I190820 06:51:20.026753 430128 storage/client_test.go:495  gossip network initialized
I190820 06:51:20.027647 430128 storage/replica_command.go:1263  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I190820 06:51:20.029045 430128 storage/replica_raft.go:291  [s1,r1/1:/M{in-ax},txn=0af6926a] proposing ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I190820 06:51:20.032437 430128 storage/store_snapshot.go:995  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 548aaee1 at applied index 18
I190820 06:51:20.032668 430128 storage/store_snapshot.go:1038  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2: kv pairs: 55, log entries: 0, rate-limit: 8.0 MiB/sec, 0.00s
I190820 06:51:20.034197 430654 storage/replica_raftstorage.go:808  [s2,r1/2:{-}] applying LEARNER snapshot [id=548aaee1 index=18]
I190820 06:51:20.034745 430284 storage/replica_command.go:1263  [replicate,s1,r1/1:/M{in-ax}] change replicas (add [] remove [(n2,s2):2LEARNER]): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
W190820 06:51:20.035114 430654 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
I190820 06:51:20.035419 430654 storage/replica_raftstorage.go:829  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=1ms ingestion=4@1ms id=548aaee1 index=18]
I190820 06:51:20.054182 430284 storage/replica_raft.go:291  [replicate,s1,r1/1:/M{in-ax},txn=2361e7dc] proposing REMOVE_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1] next=3
I190820 06:51:20.055148 430128 storage/replica_command.go:948  [s1,r1/1:/M{in-ax}] could not promote [n2,s2] to voter, rolling back: change replicas of r1 failed: descriptor changed: [expected] r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1] != [actual] r1:/M{in-ax} [(n1,s1):1, next=3, gen=2]
    --- FAIL: TestRangeTransferLeaseExpirationBased/DrainTransfer (0.15s)
        client_replica_test.go:563: change replicas of r1 failed: descriptor changed: [expected] r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1] != [actual] r1:/M{in-ax} [(n1,s1):1, next=3, gen=2]

```

--- a/pkg/storage/client_test.go
+++ b/pkg/storage/client_test.go
@@ -1166,7 +1166,7 @@ func (m *multiTestContext) changeReplicas(
 		// We can't use storage.IsSnapshotError() because the original error object
 		// is lost. We could make a this into a roachpb.Error but it seems overkill
 		// for this one usage.
-		if testutils.IsError(err, "snapshot failed: .*") {
+		if testutils.IsError(err, "snapshot failed: .*|descriptor changed") {
 			log.Info(ctx, err)
 			continue
 		}

